packages/apollo-shared/src/Checks/CDSCheck.ts

```diff
@@ -1,277 +1,229 @@
-//   if (sequence.length % 3 === 0) {
-//     const lastCodon = codons.pop() // Last codon is supposed to be a stop
-//     if (!lastCodon) {
-//       throw new Error(`No sequence found for feature "${_id}"`)
-//     }
-//     if (!(lastCodon.toUpperCase() in STOP_CODONS)) {
-//       checkResults.push({
-//         _id: new ObjectID().toHexString(),
-//         name: 'MissingStopCodonCheck',
-//         ids,
-//         refSeq: refSeq.toString(),
-//         start: end,
-//         end,
-//         message: `Feature "${_id}" is missing a stop codon`,
-//       })
-//     }
-//   } else {
-//     checkResults.push({
-//       _id: new ObjectID().toHexString(),
-//       name: 'MultipleOfThreeCheck',
-//       ids,
-//       refSeq: refSeq.toString(),
-//       start,
-//       end,
-//       message: `The coding sequence for feature "${_id}" is not a multiple of three`,
-//     })
-//   }
-//   for (const [idx, codon] of codons.entries()) {
-//     const [codonStart, codonEnd] = getOriginalCodonLocation(feature, idx)
-//     if (codon.toUpperCase() in STOP_CODONS) {
-//       checkResults.push({
-//         _id: new ObjectID().toHexString(),
-//         name: 'InternalStopCodonCheck',
-//         ids,
-//         refSeq: refSeq.toString(),
-//         start: codonStart,
-//         end: codonEnd,
-//         message: `The coding sequence for feature "${_id}" has an internal stop codon`,
-//       })
-//     }
-//   }
-//   return checkResults
-// }
+function getOriginalCodonLocation(
+  cdsLocations: CDSLocation[],
+  index: number,
+): [number, number] {
+  let lengthToStart = index * 3
+  let lengthToEnd = lengthToStart + 3
+  let startLocation: number | undefined = undefined,
+    endLocation: number | undefined = undefined
+  for (const loc of cdsLocations) {
+    const locLength = loc.max - loc.min
+    if (startLocation === undefined && locLength > lengthToStart) {
+      startLocation = loc.min + lengthToStart
+    } else {
+      lengthToStart -= locLength
+    }
+    if (endLocation === undefined && locLength > lengthToEnd) {
+      endLocation = loc.min + lengthToEnd
+    } else {
+      lengthToEnd -= locLength
+    }
+    if (startLocation !== undefined && endLocation !== undefined) {
+      return [startLocation, endLocation]
+    }
+  }
+  throw new Error(`Could not find original codon location for index ${index}`)
+}
 
 export class CDSCheck extends Check {
   name = 'CDSCheck'
   version = 1
   default = true
 
   async checkFeature(
-    _feature: AnnotationFeatureSnapshot,
-    _getSequence: (start: number, end: number) => Promise<string>,
+    _featureSnapshot: AnnotationFeatureSnapshot,
+    getSequence: (start: number, end: number) => Promise<string>,
+    feature?: AnnotationFeature,
   ): Promise<CheckResultSnapshot[]> {
-    await Promise.resolve()
-    return []
-    // if (feature.type === 'CDS') {
-    //   return checkCDS(feature, getSequence)
-    // }
+    if (!feature) {
+      return []
+    }
```

Comment:
We need to operate on `featureSnapshot` for this check, since it has to run in the back end, too. You can probably copy some of the code, e.g. `cdsLocations`, over here to help, but you'll have to adapt it to work on a `featureSnapshot` instead.